### DISCUSSION TITLE
Dynamic API endpoints responding with 404 error

### DISCUSSION THREAD

**Discussion by kagithapk** [vercel/next.js#37849] [Help]
I've created API routes `pages/api/test/index.ts` and `pages/api/test/[id].ts`.
And the endpoints are `/api/test` and `/api/test/1`

___Followed option 2 model from the [docs](https://nextjs.org/docs/api-routes/dynamic-api-routes#index-routes-and-dynamic-api-routes).___

It is working as expected in the local (`npm run dev`).
But it is not working as expected when `deployed in Vercel`.

**Expected:**
`/api/test/1` should respond with `pages/api/test/[id].ts` logic

**Behaviour:** 
`/api/test/1` is responding with `HTTP status code 404`.

Am I missing any configuration?

**Comment by icyJoseph** [comment]
Got a repo to take a look at? A deployed URL?

**Reply by nick-tap** [reply]
@icyJoseph @kagithapk were either of you able to resolve this issue? What was the solution?

**Reply by nick-tap** [reply]
If it helps anyone else, the issue was that I had another directory at the root of my project called `api` and this was clashing with the `/pages/api` directory. After renaming the root `api` directory to something else the dynamic routing started working again.
